src/components/Table.tsx

```diff
@@ -81,42 +88,93 @@ const renderField = (row: TableDataRow, column: TableColumn) => {
 	return value == null ? '' : value;
 };
 
-export default class Table extends React.Component<
-	TableProps,
+export default class Table<T> extends React.Component<
+	TableProps<T>,
 	{
-		sortColumn: null | string;
+		allChecked: boolean;
 		reverse: boolean;
+		checkedItems: T[];
+		sortColumn: null | keyof T;
 	}
 > {
-	constructor(props: TableProps) {
+	constructor(props: TableProps<T>) {
 		super(props);
 
+		if (props.onChecked && !props.rowKey) {
+			throw new Error(
+				'A `rowKey` property must be provided if using `onChecked` with a Table component',
+			);
+		}
+
 		this.state = {
-			sortColumn: null,
+			allChecked: false,
 			reverse: false,
+			checkedItems: [],
+			sortColumn: null,
 		};
 	}
 
-	sortData(data: TableDataRow[]): TableDataRow[] {
+	isChecked(item: T) {
+		const rowKey = this.props.rowKey;
+		if (!rowKey) {
+			return false;
+		}
+
+		const identifier = item[rowKey];
+		return !!find(this.state.checkedItems, x => x[rowKey] === identifier);
+	}
+
+	sortData(data: T[]): T[] {
 		if (this.state.sortColumn === null) {
 			return data;
 		}
 
-		const collection = (sortBy(data.slice(), item => {
-			const sortableValue = item[this.state.sortColumn!];
+		const collection = sortBy(data.slice(), item => {
+			const sortableValue = item[this.state.sortColumn!] as any;
 			return isPlainObject(sortableValue)
 				? (sortableValue as any).value
 				: sortableValue;
-		}) as any) as TableDataRow[];
+		});
 
 		if (this.state.reverse) {
 			reverse(collection);
 		}
 
-		return collection;
+		return (collection as any) as T[];
```

Comment:
I think that we can get rid of this casting if we use `sortBy<T>`.

Reply:
Nice, I thought there was a simple solution here :)